### PR TITLE
HPCC-28031 Add FluentUI v9 Support

### PR DESCRIPTION
**Pull request by GordonSmith**
Add matching theme support

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/layouts/MessageBox.tsx [hunk 1]
```diff
@@ -56,7 +56,7 @@ export const MessageBox: React.FunctionComponent<MessageBoxProps> = ({
 
     const close = React.useCallback(() => setShow(false), [setShow]);
 
-    return <Modal isOpen={show} onDismiss={close} isModeless={true} dragOptions={dragOptions}
+    return <Modal isOpen={show} onDismiss={close} isModeless={false} dragOptions={dragOptions}
```
Comment (jeclrsg) [inline esp/src/src-react/layouts/MessageBox.tsx:59]: Changing this modeless prop to true might have been for a specific request that allowed selecting text behind the Modal? Eg, copying a WU# to paste in a filter. (https://github.com/hpcc-systems/HPCC-Platform/pull/15487)

File: esp/src/src-react/theme-shims/themeDuplicates.ts [hunk 2]
```diff
@@ -0,0 +1,1073 @@
+/**
```
Comment (jeclrsg) [inline esp/src/src-react/theme-shims/themeDuplicates.ts:1]: Glad you added this comment, I was just about to ask if these were different somehow from those shims that were in Fluent.

File: esp/src/src-react/themes.ts [hunk 3]
```diff
@@ -58,4 +61,84 @@ export const darkTheme: PartialTheme = {
         black: "#ffffff",
         white: "#000000",
     }
-};
\ No newline at end of file
+};
+
+const brandWeb: BrandVariants = {
+    10: "#061724",
```
Comment (jeclrsg) [inline esp/src/src-react/themes.ts:67]: Where do these hexcodes come from, the older PartialThemes in themes.ts?

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith Only thing I noticed that might be a problem was that isModeless prop in MessageBox.tsx. You'd changed that to true several months back, but I don't recall for what specific reason.

File: esp/src/src-react/layouts/MessageBox.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/layouts/MessageBox.tsx:59]: I will have to take another look as bumping the @fluentui/react version with this set to true is breaking all the MessageBox instances...

File: esp/src/src-react/themes.ts (hunk 3 above)
Comment (GordonSmith) [reply esp/src/src-react/themes.ts:67]: I have added a README.md to this folder with some useful links.

File: esp/src/src-react/layouts/MessageBox.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/layouts/MessageBox.tsx:72]: Ok - found reference(s) to the issue and a workaround

Comment (jeclrsg) [reply esp/src/src-react/layouts/MessageBox.tsx:72]: The workaround was to allow for isModeless=true?

Comment (GordonSmith) [reply esp/src/src-react/layouts/MessageBox.tsx]: I have renabled it now.

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**